src/com/owncloud/android/ui/activity/ShareActivity.java

```diff
@@ -116,25 +116,42 @@ protected void onNewIntent(Intent intent) {
         }
     }
 
-    private void doShareWith(String shareeName, boolean isGroup) {
+    private void doShareWith(String shareeName, String dataAuthority) {
+
+        int lastPointIndex = dataAuthority.lastIndexOf(".");
+        String type = dataAuthority.substring(lastPointIndex + 1);
+        ShareType shareType = null;
+        switch (type) {
```

Comment:
I'd prefer not, that's more hideous to read than a switch block, and this block is not so long.